dali/pipeline/operator/eager_operator.h

```diff
@@ -177,8 +189,9 @@ class DLL_PUBLIC EagerOperator {
   std::string name_;
   std::unique_ptr<OperatorBase> op_;
 
-  static CUDAStreamLease shared_cuda_stream;
-  static std::unique_ptr<ThreadPool> shared_thread_pool;
+  static CUDAStreamLease shared_cuda_stream_;
```

Comment:
More of a question: when and where do we get this from? Does it interact with CUDA automatically?

Reply:
Added lazy creation similar to thread pool, but bear in mind that very soon it will be replaces by the proper stream handler for eager ops.

Reply:
I mean it may be entirely unnecessary, I am not sure if default initializing it really touched CUDA, it was a question. :)
Maybe @klecki or @mzient have better insight here.

Reply:
Let's agree that it will be sorted out with the exposure of streams and for now I would say it is not worse. By the way it had to touch CUDA at some point because GPU eager operators work at least in a basic case e.g. `eager.crop`.